File: spring-web-java/src/main/java/spring/web/java/domain/NotificationReception.java
package spring.web.java.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class NotificationReception extends Base {

    @Id
    @GeneratedValue
    @Column(name = "notification_reception_id")
    private Long id;
    private Boolean isRead;
    private LocalDateTime readDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "notification_id")
    private Notification notification;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member member;
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/NotificationReception.kt
package spring.web.kotlin.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
class NotificationReception private constructor(
    @Id
    @GeneratedValue
    @Column(name = "notification_reception_id")
    val id: Long?,
    val isRead: Boolean,
    val readDate: LocalDateTime,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "notification_id")
    val notification: Notification,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    val member: Member
) : Base()